src/reader/mod.rs

```diff
@@ -458,8 +459,36 @@ pub struct Reader<R> {
     parser: Parser,
 }
 
-/// Builder methods
+/// Methods specialized to reading from a slice
+impl<'a> Reader<&'a [u8]> {
```

Comment:
While it makes sense to have it in the other file, in practice it is somewhat annoying for these constructors to be in separate places.  Sticking them in the same file eliminates a couple of imports.